recipe/meta.yaml

```diff
@@ -1,63 +1,71 @@
-{% set version = "0.18.0" %}
+{% set version = "0.21.1" %}
 
 package:
   name: cartopy
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/C/Cartopy/Cartopy-{{ version }}.tar.gz
-  sha256: 7ffa317e8f8011e0d965a3ef1179e57a049f77019867ed677d49dcc5c0744434
-  patches:
-    - port310.patch
+  sha256: 89d5649712c8582231c6e11825a04c85f6f0cee94dbb89e4db23eabca1cc250a
 
 build:
-  skip: True  # [win and vc<14]
-  error_overlinking: true
-  number: 2
+  # s390x is missing geos and this package is not in the SoW
+  skip: True  # [(win and vc<14) or s390x or py<38]
+  number: 0
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - patch  # [not win]
-    - m2-patch  # [win]
   host:
     - python
     - pip
-    - cython
-    - numpy
-    - msinttypes  # [win and vc<14]
-    # proj and geos are needed as dependencies even though not listed upstream
-    - proj <8.0.0  # newer versions removed required proj_api.h
+    - setuptools
+    - setuptools_scm
+    - wheel
+    - cython 0.29
+    - numpy   1.19  # [py<310]
+    - numpy   1.21  # [py==310]
+    - numpy   1.23  # [py>=311]
     - geos
   run:
     - python
     - {{ pin_compatible('numpy') }}
-    - shapely >=1.5.6
-    - pyshp >=1.1.4
-    - six >=1.3.0
-    # proj and geos are needed as dependencies even though not listed upstream
-    - proj <8.0.0  # newer versions removed required proj_api.h
-    - geos
+    - shapely >=1.6.4
+    - pyshp >=2.1
+    - geos >=3.7.2
     # including matplotlib-base even though it's optional upstream
-    - matplotlib-base >=1.5.1
-    - scipy >=0.10
+    - matplotlib-base >=3.1
+    - scipy >=1.3.1
+    - pyproj >=3.0.0
 
 test:
-  imports:
-    - cartopy
-    - cartopy.mpl.geoaxes
-    - cartopy.crs
+  requires:
+    - pip
+    - pytest
   commands:
     - python -c "import cartopy.crs as ccrs; import matplotlib.pyplot as plt; ax = plt.axes(projection=ccrs.PlateCarree())"
-
+    - pip check
+    - python -m pytest --pyargs --doctest-modules cartopy  # [not win]
+    # Windows tests fail in several places due to path seperators 
+    #  ie: 
+    # Expected:
+    #   {config[data_dir]}/shapefiles/gshhs/{scale}/GSHHS_{scale}_L{level}.shp
+    # Got:
+    #   {config[data_dir]}\shapefiles\gshhs\{scale}\GSHHS_{scale}_L{level}.shp
```

Comment:
I think this comment can be removed. It won't help the future us as package builders.

Reply:
imho I think we should be adding comments when we are disabling or otherwise ignore unit test outcomes. This gives some information for the future package builder on what to look for in the tests, which are expected failures, and which could potentially be re-enabled. This was an attempt to provide some, context, perhaps it could be improved.